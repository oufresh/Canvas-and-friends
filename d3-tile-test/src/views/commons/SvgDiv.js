//@flow
import React from "react";

const SvgDiv = ({
  top,
  left,
  width,
  height,
  backgroundColor,
  children
}: {
  top?: number,
  left?: number,
  width: number,
  height: number,
  backgroundColor?: string,
  children: any
}) => {
  const style = {
    position: "absolute",
    top: (top ? top : 0) + "px",
    left: (left ? left : 0) + "px",
    overflow: "hidden",
    pointerEvents: "all",
    width: width + "px",
    height: height + "px",
    backgroundColor: backgroundColor ? backgroundColor : "rgba(0,0,0,1)"
  };
  return <div style={style}>{children}</div>;
};

export { SvgDiv };
