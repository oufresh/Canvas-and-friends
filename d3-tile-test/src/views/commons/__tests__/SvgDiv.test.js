import React from "react";
import { shallow } from "enzyme";
import { SvgDiv } from "../SvgDiv";

describe("SvgDiv", () => {
  it("renders without a crash ", () => {
    const component = shallow(
      <SvgDiv width={100} height={200}>
        <h1>Hello!!</h1>
      </SvgDiv>
    );
    expect(component).toMatchSnapshot();
  });
});
