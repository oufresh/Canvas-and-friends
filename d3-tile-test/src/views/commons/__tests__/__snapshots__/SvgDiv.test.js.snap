// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgDiv renders without a crash  1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,1)",
      "height": "200px",
      "left": "0px",
      "overflow": "hidden",
      "pointerEvents": "all",
      "position": "absolute",
      "top": "0px",
      "width": "100px",
    }
  }
>
  <h1>
    Hello!!
  </h1>
</div>
`;
